code/modules/power/lighting.dm

```diff
@@ -394,25 +406,53 @@
 	var/area/A = get_area(src)
 	return A.lightswitch && A.power_light
 
-/obj/machinery/light/proc/flicker(amount = rand(10, 20))
-	if(flickering)
+///flicker lights on and off
+/obj/machinery/light/proc/flicker(toggle_flicker = FALSE)
+	if(!has_power())
+		addtimer(CALLBACK(src, PROC_REF(flicker)), flicker_time)
 		return
```

Comment:
Why this? If it loses power it's going to just keep calling this proc for the rest of time. This should be hooked into whatever proc it is that turns the light back on.